src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,161 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (isIndexOutForAddMethod(index)) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node<T> newNode = new Node<>(null, value, null);
+        if (index == 0) {
+            newNode.next = head;
+            head = newNode;
+        } else {
+            Node<T> current = getNodeForAddAndRemoveMethods(index);
+            newNode.next = current.next;
+            current.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isIndexExists(index)) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node<T> current = getNodeForSetAndGetMethods(index);
+
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (isIndexExists(index)) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node<T> current = getNodeForSetAndGetMethods(index);
+
+        T oldValue = current.value;
+        current.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
```

Comment:
should look smth like this

        checkIndex(index);
        return unlink(getNode(index));

Reply:
you should create an `unlink` method, as mentioned in the common mistakes checklist, and use it in remove methods

Reply:
https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md